packs/hpe-icsp/actions/icsp_server_attributes_get.py

```diff
@@ -0,0 +1,60 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from lib.icsp import ICSPBaseActions
+
+
+class GetServerAttributes(ICSPBaseActions):
+    def run(self, mids, connection_details,
+            attribute_key, attribute_type="all"):
+        if connection_details:
+            self.setConnection(connection_details)
+        self.getSessionID()
+        endpoint = "/rest/os-deployment-servers"
+        results = {}
+        for mid in mids:
```

Comment:
Probably a good idea to refactor this on a utility method on a base class since it's repeated across multiple actions.